code/modules/shuttles/shuttle_arrival.dm

```diff
@@ -20,10 +20,35 @@
 			return	//someone cancelled the launch
 
 		if (at_station() && forbidden_atoms_check())
-			//cancel the launch because of forbidden atoms. announce over supply channel?
+			//cancel the launch because of forbidden atoms
 			moving_status = SHUTTLE_IDLE
 			global_announcer.autosay("Unacceptable items detected aboard the arrivals shuttle. Launch attempt failed. Restarting launch in one minute.", "Arrivals Shuttle Oversight")
 			SSarrivals.set_launch_countdown(60)
+			SSarrivals.failreturnnumber++
+			if(SSarrivals.failreturnnumber >= 2) // get off my shuttle fool
+				var/list/mobstoyellat = mobs_in_area(get_location_area(location))
+				if (!mobstoyellat || !mobstoyellat.len)
+					return
+				for(var/mob/living/A in mobstoyellat)
+					A <<"<span class='danger'>You feel as if you shouldn't be on the shuttle.</span>" // give them an angry text
+					if(!A.client && ishuman(A) && SSarrivals.failreturnnumber >= 3) // well they are SSD and holding up the shuttle so might as well.
+						SSjobs.DespawnMob(A)
+						global_announcer.autosay("[A.real_name], [A.mind.role_alt_title], has entered long-term storage.", "Cryogenic Oversight")
+						mobstoyellat -= A // so they don't get told on
+					else if(A.client && ishuman(A) && SSarrivals.failreturnnumber >= 5 && A.newarrival == 1) // they aren't SSD and are holding up the shuttle so we are booting them. Only if they just came on the shuttle though.
```

Comment:
This is not robust, specifically, the `newarrival` check. If a player somehow delays himself getting onto the shuttle long enough, then he'll not be despawned. Either make a Could somehow make SSarrivals itself track who's been counted how long by using a persistent list instead of nuking `mobs_to_yell_at`  every time.

So basically:
* A static list is compiled the first time it stops. Every mob is given a value of 1 by doing `static_mobs_list[M] = 1`
* Every other failure time, `static_mobs_list[M]++`
* Then you use `static_mobs_list[M] > n` to determine if they should be despawned or fucked with.

Though there is a danger of it catching mobs that have been on the station and walked onto the shuttle. Hrm. Maybe not use a timer for the removal of `newarrival` and instead use an area/Exited() proc?